bunkatopics/_bunkatopics.py

```diff
@@ -25,6 +25,14 @@
 from sklearn.preprocessing import MinMaxScaler
 from tqdm import tqdm
 
+try:
+    from cuml.manifold.umap import UMAP as cuUMAP
+
+    CUML_AVAILABLE = True
+except ImportError:
+    cuUMAP = None
+    CUML_AVAILABLE = False
```

Comment:
can we have a larger type of error, for instance, when running on my Mac M1 where I don't have CUML, I get the following mistake:

---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
Cell In[9], [line 2](vscode-notebook-cell:?execution_count=9&line=2)
      [1](vscode-notebook-cell:?execution_count=9&line=1) try:
----> [2](vscode-notebook-cell:?execution_count=9&line=2)     from cuml.manifold.umap import UMAP as cuUMAP
      [4](vscode-notebook-cell:?execution_count=9&line=4)     CUML_AVAILABLE = True
      [5](vscode-notebook-cell:?execution_count=9&line=5) except ImportError:

File ~/.cache/pypoetry/virtualenvs/bunkatopics-ybLbeXnh-py3.10/lib/python3.10/site-packages/cuml/__init__.py:[1](https://vscode-remote+ssh-002dremote-002b51-002e159-002e168-002e232.vscode-resource.vscode-cdn.net/home/charles/bunka_research/bunka_performances/projection/~/.cache/pypoetry/virtualenvs/bunkatopics-ybLbeXnh-py3.10/lib/python3.10/site-packages/cuml/__init__.py:1)7
      1 #
      [2](https://vscode-remote+ssh-002dremote-002b51-002e159-002e168-002e232.vscode-resource.vscode-cdn.net/home/charles/bunka_research/bunka_performances/projection/~/.cache/pypoetry/virtualenvs/bunkatopics-ybLbeXnh-py3.10/lib/python3.10/site-packages/cuml/__init__.py:2) # Copyright (c) 2022-2023, NVIDIA CORPORATION.
      [3](https://vscode-remote+ssh-002dremote-002b51-002e159-002e168-002e232.vscode-resource.vscode-cdn.net/home/charles/bunka_research/bunka_performances/projection/~/.cache/pypoetry/virtualenvs/bunkatopics-ybLbeXnh-py3.10/lib/python3.10/site-packages/cuml/__init__.py:3) #
   (...)
     [14](https://vscode-remote+ssh-002dremote-002b51-002e159-002e168-002e232.vscode-resource.vscode-cdn.net/home/charles/bunka_research/bunka_performances/projection/~/.cache/pypoetry/virtualenvs/bunkatopics-ybLbeXnh-py3.10/lib/python3.10/site-packages/cuml/__init__.py:14) # limitations under the License.
     [15](https://vscode-remote+ssh-002dremote-002b51-002e159-002e168-002e232.vscode-resource.vscode-cdn.net/home/charles/bunka_research/bunka_performances/projection/~/.cache/pypoetry/virtualenvs/bunkatopics-ybLbeXnh-py3.10/lib/python3.10/site-packages/cuml/__init__.py:15) #
---> [17](https://vscode-remote+ssh-002dremote-002b51-002e159-002e168-002e232.vscode-resource.vscode-cdn.net/home/charles/bunka_research/bunka_performances/projection/~/.cache/pypoetry/virtualenvs/bunkatopics-ybLbeXnh-py3.10/lib/python3.10/site-packages/cuml/__init__.py:17) from cuml.internals.base import Base, UniversalBase
     [18](https://vscode-remote+ssh-002dremote-002b51-002e159-002e168-002e232.vscode-resource.vscode-cdn.net/home/charles/bunka_research/bunka_performances/projection/~/.cache/pypoetry/virtualenvs/bunkatopics-ybLbeXnh-py3.10/lib/python3.10/site-packages/cuml/__init__.py:18) from cuml.internals.available_devices import is_cuda_available
     [20](https://vscode-remote+ssh-002dremote-002b51-002e159-002e168-002e232.vscode-resource.vscode-cdn.net/home/charles/bunka_research/bunka_performances/projection/~/.cache/pypoetry/virtualenvs/bunkatopics-ybLbeXnh-py3.10/lib/python3.10/site-packages/cuml/__init__.py:20) # GPU only packages

File ~/.cache/pypoetry/virtualenvs/bunkatopics-ybLbeXnh-py3.10/lib/python3.10/site-packages/cuml/internals/__init__.py:[1](https://vscode-remote+ssh-002dremote-002b51-002e159-002e168-002e232.vscode-resource.vscode-cdn.net/home/charles/bunka_research/bunka_performances/projection/~/.cache/pypoetry/virtualenvs/bunkatopics-ybLbeXnh-py3.10/lib/python3.10/site-packages/cuml/internals/__init__.py:1)8
      1 #
      [2](https://vscode-remote+ssh-002dremote-002b51-002e159-002e168-002e232.vscode-resource.vscode-cdn.net/home/charles/bunka_research/bunka_performances/projection/~/.cache/pypoetry/virtualenvs/bunkatopics-ybLbeXnh-py3.10/lib/python3.10/site-packages/cuml/internals/__init__.py:2) # Copyright (c) 2019-2023, NVIDIA CORPORATION.
      [3](https://vscode-remote+ssh-002dremote-002b51-002e159-002e168-002e232.vscode-resource.vscode-cdn.net/home/charles/bunka_research/bunka_performances/projection/~/.cache/pypoetry/virtualenvs/bunkatopics-ybLbeXnh-py3.10/lib/python3.10/site-packages/cuml/internals/__init__.py:3) #
   (...)
     [14](https://vscode-remote+ssh-002dremote-002b51-002e159-002e168-002e232.vscode-resource.vscode-cdn.net/home/charles/bunka_research/bunka_performances/projection/~/.cache/pypoetry/virtualenvs/bunkatopics-ybLbeXnh-py3.10/lib/python3.10/site-packages/cuml/internals/__init__.py:14) # limitations under the License.
...
--> [374](https://vscode-remote+ssh-002dremote-002b51-002e159-002e168-002e232.vscode-resource.vscode-cdn.net/usr/lib/python3.10/ctypes/__init__.py:374)     self._handle = _dlopen(self._name, mode)
    [375](https://vscode-remote+ssh-002dremote-002b51-002e159-002e168-002e232.vscode-resource.vscode-cdn.net/usr/lib/python3.10/ctypes/__init__.py:375) else:
    [376](https://vscode-remote+ssh-002dremote-002b51-002e159-002e168-002e232.vscode-resource.vscode-cdn.net/usr/lib/python3.10/ctypes/__init__.py:376)     self._handle = handle

OSError: libcudart.so: cannot open shared object file: No such file or directory
Output is truncated. View as a [scrollable element](command:cellOutput.enableScrolling?0a37ec37-8272-4f3a-ae55-71480d314404) or open in a [text editor](command:workbench.action.openLargeOutput?0a37ec37-8272-4f3a-ae55-71480d314404). Adjust cell output [settings](command:workbench.action.openSettings?%5B%22%40tag%3AnotebookOutputLayout%22%5D)...

Reply:
done

Reply:
flake8 does not accept bare 'except', is it possible to add at least OSError or any other one ?

Reply:
should be ok